dexios/src/subcommands/header.rs

```diff
@@ -126,13 +126,39 @@ pub fn restore(input: &str, output: &str, skip: SkipMode) -> Result<()> {
         exit(1);
     }
 
-    let (header, _) = header_result.context("Error unwrapping the header's result")?; // this should never happen
+    let (header, _) = header_result.context("Error unwrapping the header's result")?;
 
     let mut output_file = OpenOptions::new()
+        .read(true)
         .write(true)
         .open(output)
         .with_context(|| format!("Unable to open output file: {}", output))?;
 
+    // TODO(brxken128): convert this to a match statement, (V4 and below write directly, V5 and above check for empty space beforehand)
+    if header.header_type.version >= HeaderVersion::V5 {
+        let mut expected_empty_space = vec![
+            0u8;
+            header
+                .get_size()
+                .try_into()
+                .context("Unable to get header's size as usize")?
+        ];
+        output_file
+            .read_exact(&mut expected_empty_space)
+            .context("Unable to read the start of the file")?;
+        output_file
+            .seek(std::io::SeekFrom::Current(
+                -i64::try_from(header.get_size()).context("Unable to get header's size as i64")?,
+            ))
+            .context("Unable to seek back to start of file")?;
+
+        if expected_empty_space.is_empty() {
+            header.write(&mut output_file)?;
+        } else {
+            return Err(anyhow::anyhow!("No empty space found at the start of {}! It's either not an encrypted file, or it was encrypted in detached mode (and the header can't be restored)", output));
```

Comment:
@pleshevskiy I think this is a good compromise. Encrypting in detached mode removes the empty space from the start of the file (and makes restoring a lot harder), but *stripping* the header leaves that empty space intact, so we can restore. I'm going to tweak the code and such so that it's easier on the eyes :)

Reply:
But there is no more empty space in the new files encrypted by new detached mode) you will have an error) You can use [my solution](https://github.com/brxken128/dexios/blob/master/dexios/src/domain/decrypt.rs#L62)

Reply:
> But there is no more empty space in the new files encrypted by detached mode) you will have an error) You can use [my solution](https://github.com/brxken128/dexios/blob/master/dexios/src/domain/decrypt.rs#L62)

The point was to error :)

My plan is to entirely disallow restoring encrypted/detached headers entirely, but we can still restore `stripped` headers! Stripping just replaces the header with 0s, so the empty space is still present for us to write to.

If a user encrypts in detached mode, it's almost certain that they know what they're doing. There's more mystery surrounding how the files were encrypted and stuff, but they lose access to re-attaching the header to the file. We can add notes about this in the docs, and in the man page (when I get around to making one).

Reply:
As far as I know it will break on `read_exact` ) So my way is still better) and there you can call the error you want.)